src/Commands/BuildToolsBase.php

```diff
@@ -1227,6 +1234,59 @@ protected function exec($command)
         return $outputLines;
     }
 
+    /**
+     * Download a copy of the secrets.json file from the appropriate site.
+     */
+    protected function downloadSecrets($site_env_id, $filename)
+    {
+        $workdir = $this->tempdir();
+        $this->rsync($site_env_id, ":files/private/$filename", $workdir, true);
+
+        if (file_exists("$workdir/$filename"))
+        {
+            $secrets = file_get_contents("$workdir/$filename");
+            $secretValues = (array)json_decode($secrets);
```

Comment:
Use `json_decode($secrets, true);`. Passing `true` will make json_decode convert all elements in the structure to arrays. Casting to `(array)` only does the top level, and leaves everything else stdclass.